agents/taskComplexity.js

```diff
@@ -4,7 +4,7 @@ async function taskComplexityAgent(summaries, task) {
     const prompt = 
 ` 
 USER INPUT: ${task}
-YOUR TASK: You are a project manager working in a software development project. Measure the complexity of doing the USER INPUT and decide if it's critical to breakdown the work into less complex JIRA tasks. Based on your assessment output the list of tasks. Tasks will be done by engineers
+YOUR TASK: You are a project manager working in a software development project. Measure the complexity of doing the USER INPUT and decide if it's critical to breakdown the work into less complex JIRA tasks. Based on your assessment write at least 1 task. Tasks will be done by engineers. Resources are limited so only divide tasks if crucial.
```

Comment:
Depending on the overall flow plan, I think we would want to create the maximum possible tasks that still fit the possible text size. The simpler the tasks we know the better they are going to be executed. Also it might be easier for a human to review the top tasks...

Just thinking out loud and maybe for the longer term.

I think this agent by itself is a good point to start doing some kind of automated testing, the problem is the result is unpredictable and inconsistent. 
But I'd love to see test cases and a testing script.
the testing cases would start to describe what we know/expect to work.

well maybe instead of the standard result testing, we can have another ai call review the result and answer a few points to grade it on:
* Complexity
* Ease of execution for a junior developer

something like that

as to know if we did a good job here...

Reply:
Yap, agreed. This is a step in adding a new agent into the loop. Not perfect tho.

Output Quality:
We don't know what works and what doesn't. So we need a benchmarking process. Something that we can use to assess the quality of the output, which will help us know if we are improving autopilot or not.

Agent's Manager/Reviewer:
We should add a new agent that reviews the work of agents and accepts it or asks the agent to try again. We know that GPT sometimes needs multiple tries.